system_zero/src/env/environment.rs

```diff
@@ -0,0 +1,28 @@
+use primitive_types::{H160, H256, U256};
+
+/// An interface to the blockchain environment, for reading and writing various chain data.
+pub trait Environment {
```

Comment:
cc @BGluth, let me know if you have any concerns on this part, since it relates to the client (I imagine we'd have a sled-backed `Environment` impl in mir-core).

Reply:
This looks good!

My only thought is whether or not we need a function to create an EOA as well.

Reply:
I was thinking a transfer could be implemented as a `sub_balance` (which I forgot to add) followed by an `add_balance`. I believe that should work for EOA transfers as well as transfers triggered from the EVM (e.g. `CALL` with a nonzero value). Does that sound okay?

We could alternatively do it as a `transfer` function, but I figure `sub_balance`/`add_balance` could have other uses, like charging gas.

Reply:
Right. I guess there's no real work to do for a transfer, so there's no need to explicitly create an EOA. Yeah this seems good.